### PR TITLE
[Doc] Update debugging docs with YellowBox/RedBox

### PR DESCRIPTION
**Pull request by cosmith**
See https://github.com/facebook/react-native/issues/5271

### CODE REVIEW

**Comment by christopherdro** [comment]
@cosmith Thanks for the PR>

Any change you can rebase with master so we can merge this in?
Thanks!

File: docs/Debugging.md [hunk 1]
```diff
@@ -20,9 +20,19 @@ To access the in-app developer menu:
 > 1. For iOS open your project in Xcode and select `Product` → `Scheme` → `Edit Scheme...` (or press `⌘ + <`). Next, select `Run` from the menu on the left and change the Build Configuration to `Release`.
 > 2. For Android, by default, developer menu will be disabled in release builds done by gradle (e.g with gradle `assembleRelease` task). Although this behavior can be customized by passing proper value to `ReactInstanceManager#setUseDeveloperSupport`.
 
+### Android logging
+Run `adb logcat *:S ReactNative:V ReactNativeJS:V` in a terminal to see your Android app's logs.
```
Comment (mkonicek) [inline docs/Debugging.md]: I feel like we should add a command for this, something like `react-native android log` but this is great for now. Thank you!

File: docs/Debugging.md [hunk 2]
```diff
@@ -20,9 +20,19 @@ To access the in-app developer menu:
...
+
 ### Reload
 Selecting `Reload` (or pressing `⌘ + r` in the iOS simulator) will reload the JavaScript that powers your application. If you have added new resources (such as an image to `Images.xcassets` on iOS or to `res/drawable` folder on Android) or modified any native code (Objective-C/Swift code on iOS or Java/C++ code on Android), you will need to re-build the app for the changes to take effect.
 
+### YellowBox/RedBox
+Using `console.warn` will display an on-screen log on a yellow background. Click on this warning to show more information about it full screen and/or dismiss the warning.
+
+You can use `console.error` to display a full screen error on a red background.
```
Comment (mkonicek) [inline docs/Debugging.md]: Does `console.error` also crash the app in release mode? If yes we should mention that too.

**Comment by mkonicek** [comment]
Thanks for improving the docs @cosmith!

File: docs/Debugging.md (hunk 2 above)
Comment (cosmith) [inline docs/Debugging.md]: No, calling `console.error` in release mode only shows an error message in the logs.

**Comment by mkonicek** [comment]
Cool, let's merge this.

@facebook-github-bot shipit

**Comment by christopherdro** [comment]
:+1: Thanks

**Pull request closed without merging**